### PR TITLE
Duplicate 3.2 EmailValidator to 2.8

### PR DESCRIPTION
**Pull request by trsteel88**
Allows egulias/email-validator ~2.0

| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by stof** [comment]
New features are never added in maintenance branches. This goes against semver (we don't want to release a 2.9 version to add new features in 2.x). So 👎 from me.

File: src/Symfony/Component/Validator/Constraints/EmailValidator.php [hunk 1]
```diff
@@ -56,39 +57,32 @@ public function validate($value, Constraint $constraint)
...
-                        ->setParameter('{{ value }}', $this->formatValue($value))
-                        ->setCode(Email::INVALID_FORMAT_ERROR)
-                        ->addViolation();
-                }
-
-                return;
-            }
-        } elseif (!preg_match('/^.+\@\S+\.\S+$/', $value)) {
-            if ($this->context instanceof ExecutionContextInterface) {
+            if (interface_exists(EmailValidation::class) && !$strictValidator->isValid($value, new NoRFCWarningsValidation())) {
```
Comment (stof) [inline src/Symfony/Component/Validator/Constraints/EmailValidator.php]: 2.8 is compatible with PHP 5.3, so this code cannot even work

**Comment by stof** [comment]
@trsteel88 there are other issues with your PR (breaking the backward compatibility layer for the old execution context, which cannot be done in 2.8 as it is a BC break).
But spending time on fixing this does not make sense anyway, as the PR cannot be accepted anyway (support for v2 of the library was considered as being a feature)

**Comment by trsteel88** [comment]
If egulias/email-validator ~2.0 isn't supported, shouldn't it have a ~1.2 restriction in the composer.json?

This is causing a break in an application since nothing is restricting it.

**Comment by trsteel88** [comment]
@stof I've submitted an alternative to this PR - https://github.com/symfony/symfony/pull/22724

It just restricts egulias/email-validator to <2.0

**Comment by ro0NL** [comment]
> support for v2 of the library was considered as being a feature

If we support it in a BC manner.. this shouldnt be a big deal right? Otherwise :+1: for rebasing on master :)

**Comment by xabbuh** [comment]
Closing here as this cannot be merged as explained before.

**Pull request closed without merging**